File: php/features.php
<?php
declare(strict_types=1);

// EvilInsultGenerator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EvilInsultGeneratorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EvilInsultGeneratorBaseFeature();
            case "test":
                return new EvilInsultGeneratorTestFeature();
            default:
                return new EvilInsultGeneratorBaseFeature();
        }
    }
}

File: py/features.py
# EvilInsultGenerator SDK feature factory

from feature.base_feature import EvilInsultGeneratorBaseFeature
from feature.test_feature import EvilInsultGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EvilInsultGeneratorBaseFeature(),
        "test": lambda: EvilInsultGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
